nominations/templates/admin/application_pdf.html

```diff
@@ -0,0 +1,182 @@
+        {% endif %}
+
+      </ul>
+
+      <ul>
+        {% if app.primary_election_date %}
+          <li><strong>Primary election date:</strong> {{app.primary_election_date}}</li>
+        {% endif %}
+
+        {% if app.general_election_date %}
+          <li><strong>General election date:</strong> {{app.general_election_date}}</li>
+        {% endif %}
+
+        <li><strong>Classification: </strong> Level {{app.get_classification_level_display}}</li>
+
+        <li><strong>Recommendation written by:</strong> {{app.get_staff_display}}</li>
+
+        <li><strong>Staff Recommendation:</strong> <strong>{{app.get_recommendation_display}}</strong> {{app.candidate_first_name}} {{app.candidate_last_name}} for {{app.candidate_office}}</li>
+      </ul>
+
+      {% if app.state_of_the_race %}
+        <h2>State of the Race</h2>
+        {{app.state_of_the_race|safe}}
+
+        <ul>
+          <li>
+            {% if app.vol_incumbent %}
+              <strong>Candidate {{app.vol_incumbent|yesno:"is, is not"}} an incumbent.</strong>
+            {% endif %}
+          </li>
+
+          <li>
+            {% if app.vol_dem_challenger %}
+              Dem challenger? {{app.dem_challenge|yesno:"Yes, No"}}
+            {% endif %}
+          </li>
+
+          <li>
+            {% if app.vol_other_progressives %}
+              <strong>Other progressives in race:</strong> {{app.vol_other_progressives }}
+            {% endif %}
+          </li>
+
+          <li>
+            {% if app.vol_polling %}
+              <strong>Polling:</strong> {{app.vol_polling}}
+            {% endif %}
+          </li>
+
+          <li>
+            {% if app.vol_endorsments %}
+              <strong>Endorsements:</strong> {{app.vol_endorsements}}
+            {% endif %}
+          </li>
+
+          <li>
+            {% if app.vol_advantage %}
+              <strong>Previous Election D% or R% Advantage:</strong> {{app.vol_advantage}}
+            {% endif %}
+          </li>
+
+          <li>
+            {% if app.vol_turnout %}
+              <strong>Previous Election Year Turnout:</strong> {{app.vol_turnout}}
+            {% endif %}
+          </li>
+
+          <li>
+            {% if app.vol_win_number %}
+              <strong>Win Number:</strong> {{app.vol_win_number}}
+            {% endif %}
+          </li>
+
+          <li>
+            {% if app.vol_fundraising %}
+              <strong>Fundraising:</strong> ${{app.vol_fundraising}}
+            {% endif %}
+          </li>
+
+          <li>
+            {% if app.vol_opponent_fundraising %}
+              <strong>Opponent Fundraising:</strong> ${{app.vol_opponent_fundraising}}
+            {% endif %}
+          </li>
+        </ul>
+      {% endif %}
+
+      {% if app.local_group_info %}
+        <h2>{{app.group}} Info</h2>
+        {{app.local_group_info|safe}}
```

Comment:
Why using `|safe` everywhere? What does it do?

Reply:
These are rich text fields, consisting of HTML, i.e. `<p>Some text</p>`. The `safe` filter is a django template filter that disables HTML escaping. https://docs.djangoproject.com/en/1.8/ref/templates/builtins/#safe. Without it, we'd get `<p>Some text</p>` in the PDF instead of `Some text`.